lib/Listener/RestrictStartingCalls.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2019 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Listener;
+
+
+use OCA\Talk\Exceptions\ForbiddenException;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCP\IConfig;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+class RestrictStartingCalls {
+
+	/** @var IConfig */
+	protected $config;
+
+	public function __construct(IConfig $config) {
+		$this->config = $config;
+	}
+
+	public static function register(EventDispatcherInterface $dispatcher): void {
+		$dispatcher->addListener(Room::class . '::preSessionJoinCall', function(GenericEvent $event) {
+
+			/** @var self $listener */
+			$listener = \OC::$server->query(self::class);
+			$listener->checkStartCallPermissions($event);
+		});
+	}
+
+	/**
+	 * @param GenericEvent $event
+	 * @throws ForbiddenException
+	 */
+	public function checkStartCallPermissions(GenericEvent $event): void {
+		/** @var Room $room */
+		$room = $event->getSubject();
+		/** @var string $sessionId */
+		$sessionId = $event->getArgument('sessionId');
+		/** @var Participant $participant */
+		$participant = $room->getParticipantBySession($sessionId);
+
+		$defaultStartCall = (int) $this->config->getAppValue('spreed', 'start_calls', Room::START_CALL_EVERYONE);
```

Comment:
Exactly the same code is added to `RoomController`. I suggest moving this to a new function `Room::canStartCall($participant)` to avoid duplication.

Reply:
Done